conans/client/build/cmake_flags.py

```diff
@@ -36,7 +36,8 @@ def get_generator(settings):
 
     if not compiler or not compiler_version or not arch:
         if os_build == "Windows":
-            return "MinGW Makefiles"
+            logger.warning("No compiler has been detected for Windows platform")
```

Comment:
Maybe the message should be that generator could not be deduced? Should we use this same message to raise a exception in case of calling the build() method?

Reply:
I don't know if we can raise an error, it will break some users I think. But I agree with you about the message, it is not about compiler, but generator, I need to fix it. thanks

Reply:
done